test/s3select_test.cpp

```diff
@@ -222,6 +222,86 @@ TEST(TestS3SElect, arithmetic_operator)
   ASSERT_EQ( ( (a+b) * (c+d) ).i64(), 21 );
 }
 
+TEST(TestS3SElect, intnull_compare_operator)
+{
+  value a10(10), b11(11), c10(10), d, e;
+  d.setnull();
+  e.setnull();
+  ASSERT_EQ( d > b11, false );
+  ASSERT_EQ( d >= c10, false );
+  ASSERT_EQ( d < a10, false );
+  ASSERT_EQ( d <= b11, false );
+  ASSERT_EQ( d != a10, true );
+  ASSERT_EQ( d != e, true );
+  ASSERT_EQ( d == a10, false );
+}
+
+TEST(TestS3SElect, floatnull_compare_operator)
+{
+  value a10(10.1), b11(11.2), c10(10.1), d, e;
+  d.setnull();
+  e.setnull();
+  ASSERT_EQ( d > b11, false );
+  ASSERT_EQ( d >= c10, false );
+  ASSERT_EQ( d < a10, false );
+  ASSERT_EQ( d <= b11, false );
+  ASSERT_EQ( d != a10, true );
+  ASSERT_EQ( d != e, true );
+  ASSERT_EQ( d == a10, false );
+}
+
+TEST(TestS3SElect, intnan_compare_operator)
+{
+  value a10(10), b11(11), c10(10), d, e;
+  d.set_nan();
+  e.set_nan();
+  ASSERT_EQ( d > b11, false );
+  ASSERT_EQ( d >= c10, false );
+  ASSERT_EQ( d < a10, false );
+  ASSERT_EQ( d <= b11, false );
+  ASSERT_EQ( d != a10, true );
+  ASSERT_EQ( d != e, true );
+  ASSERT_EQ( d == a10, false );
+}
+
+TEST(TestS3SElect, floatnan_compare_operator)
+{
+  value a10(10.1), b11(11.2), c10(10.1), d, e;
+  d.set_nan();
+  e.set_nan();
+  ASSERT_EQ( d > b11, false );
+  ASSERT_EQ( d >= c10, false );
+  ASSERT_EQ( d < a10, false );
+  ASSERT_EQ( d <= b11, false );
+  ASSERT_EQ( d != a10, true );
+  ASSERT_EQ( d != e, true );
+  ASSERT_EQ( d == a10, false );
+}
+
+TEST(TestS3SElect, null_arithmetic_operator)
+{
+  value a(7), d;
+  d.setnull();
+  ASSERT_EQ((a + d).to_string(), "nan" );
+  ASSERT_EQ((a - d).to_string(), "nan" );
+  ASSERT_EQ((a * d).to_string(), "nan" );
+  ASSERT_EQ((a / d).to_string(), "nan" ); 
+}
+
+TEST(TestS3SElect, nan_arithmetic_operator)
+{
+  value a(7), d;
+  d.set_nan();
+  float b = ((a + d).dbl() );
+  float c = ((a - d).dbl() );
+  float v = ((a * d).dbl() );
+  float w = ((a / d).dbl() );
+  EXPECT_TRUE(!(b <= b) ); 
```

Comment:
that's good, it demonstrates both compare and arithmetic operators.
replace the /EXPECT_TRUE(!/ 
with
/EXPECT_FALSE( b<=b) /

it's more readable.

Reply:
replace with EXCEPT_FALSE